src/tracker/tracker_sampler.jl

```diff
@@ -0,0 +1,100 @@
+abstract type AbstractROI end
+mutable struct BoxROI{T <: AbstractArray, S <: AbstractArray} <: AbstractROI
```

Comment:
I don't think we should be storing the entire image as a field in `BoxROI`. An abstract region of interest could just be a set of `AbstractRange` types, one for each dimensions. It would also not need to be mutable. You just instantiate a new when the ROI changes. I imagine something like this:

```julia
struct BoxROI{R₁ <: AbstractRange, R₂} <: AbstractROI
    span₁::R₁ 
    span₂::R₂
end
```